webpack.config.js

```diff
@@ -1,62 +1,70 @@
-const path = require('path');
-const ExtractTextPlugin = require('extract-text-webpack-plugin');
-
-const stylesCSSPlugin = new ExtractTextPlugin( {
-	filename: './styles.css',
-} );
+// WordPress dependencies
 
-const adminCSSPlugin = new ExtractTextPlugin( {
-	filename: './admin.css',
-} );
+const defaultConfig = require( '@wordpress/scripts/config/webpack.config.js' );
 
-// Configuration for the ExtractTextPlugin.
-const extractConfig = {
-	use: [
-		{ loader: 'raw-loader' },
-		{
-			loader: 'postcss-loader',
-			options: {
-				plugins: [
-					require( 'autoprefixer' ),
-				],
-			},
-		},
-		{
-			loader: 'sass-loader',
-			query: {
-				outputStyle: 'production' === process.env.NODE_ENV ? 'compressed' : 'nested',
-			},
-		},
-	],
-};
+// External dependencies
+const path = require('path');
+const webpack = require( 'webpack' );
+const MiniCssExtractPlugin = require( 'mini-css-extract-plugin' );
+const FixStyleOnlyEntriesPlugin = require( 'webpack-fix-style-only-entries' );
+const OptimizeCssAssetsPlugin = require( 'optimize-css-assets-webpack-plugin' );
 
 module.exports = {
-	entry: { main: './src/index.js' },
+	...defaultConfig,
+	devtool: 'source-map',
+	externals: {
+		jquery: 'jQuery',
+		$: 'jQuery'
+	},
+	entry: {
+		// JS
+		main: './src/index.js',
+		// SCSS
+		style: './src/styles.scss',
+	},
 	output: {
-		path: path.resolve(__dirname, 'dist'),
-		filename: 'main.js'
+		filename: 'build/[name].js',
+		path: __dirname
 	},
+	plugins: [
+		...defaultConfig.plugins,
+		new FixStyleOnlyEntriesPlugin(),
+		new OptimizeCssAssetsPlugin(),
+		new webpack.ProvidePlugin( {
+			Promise: 'es6-promise-promise',
+			$: 'jquery',
+		} ),
+		new MiniCssExtractPlugin( {
+			filename: 'dist/styles.css',
```

Comment:
```suggestion
			filename: 'build/[name].css',
```

Update to the new `/build` directory and use the placeholder `[name]` which gets replaced with the entry point name (`styles`, `admin`, etc).